### PR TITLE
Extract counter in sub model sample

### PR DESCRIPTION
**Pull request by bender2k14**

### CODE REVIEW

File: src/Samples/SingleCounter/Program.fs [hunk 1]
```diff
@@ -18,6 +18,8 @@ type Msg =
   | SetStepSize of int
   | Reset
 
+let canReset = (<>) (init ())
```
Comment (cmeeren) [inline src/Samples/SingleCounter/Program.fs]: I don't like point-free style for this free-standing function (I mostly use point-free style for inline functions). Could you write this as `let canReset m = m <> init ()` or keep inline?

File: src/Samples/SubModel.Views/SubModel.Views.csproj [hunk 2]
```diff
@@ -26,6 +26,12 @@
     <Reference Include="WindowsBase" />
   </ItemGroup>
 
+  <ItemGroup>
```
Comment (cmeeren) [inline src/Samples/SubModel.Views/SubModel.Views.csproj]: Please revert this file. VS does this automatically, but it's not needed. (.NET Core 3.0 will bring and end to the csproj workarounds.)

File: src/Samples/SubModel/Program.fs [hunk 3]
```diff
@@ -4,6 +4,41 @@ open System
...
+    { Count = 0
+      StepSize = 1 }
+  
+  type Msg =
+    | Increment
+    | Decrement
+    | SetStepSize of int
+    | Reset
+    
+  let canReset = (<>) (init ())
```
Comment (cmeeren) [inline src/Samples/SubModel/Program.fs]: Same as previous comment: Inline or write as not point-free.

File: src/Samples/SubModel/Program.fs [hunk 4]
```diff
@@ -4,6 +4,41 @@ open System
...
+    | Reset -> init ()
+  
+  let bindings () : Binding<Model, Msg> list = [
+    "CounterValue" |> Binding.oneWay (fun m -> m.Count)
+    "Increment" |> Binding.cmd Increment
+    "Decrement" |> Binding.cmd Decrement
+    "StepSize" |> Binding.twoWay(
+      (fun m -> float m.StepSize),
+      int >> SetStepSize)
+    "Reset" |> Binding.cmdIf(Reset, (<>) (init ()))
```
Comment (cmeeren) [inline src/Samples/SubModel/Program.fs]: Shouldn't this use `canReset`? (Unless inlined, of course)

File: src/Samples/SubModel/Program.fs [hunk 5]
```diff
@@ -36,39 +71,24 @@ module Clock =
...
   type Model =
-    { Count: int
-      StepSize: int
+    { Counter: Counter.Model
       Clock: Clock.Model }
 
   let init =
-    { Count = 0
-      StepSize = 1
+    { Counter = Counter.init ()
```
Comment (cmeeren) [inline src/Samples/SubModel/Program.fs]: Come to think of it, I'm thinking that since `init` is pure and does not depend on anything, perhaps it should simply be a value, not a function. What do you think? Perhaps this should be modified in all samples? (Could probably be a separate PR)

**Comment by bender2k14** [comment]
> By the way, would have been great to have this spelled out in the PR description - I had to spend a few minutes trying to figure out the point of the changes before starting my review. :) (The motivation seems obvious in hindsight, as things generally do...)

Oh, I am sorry.  I thought the title said enough.  Next time I will make sure to elaborate in the description.

> Also, it would be great in general if you could just leave a quick note on how confident you are that the changes are correct (e.g. "sample tested and works" or "untested, but compiles and seems obvious it will work"). The samples in particular are not covered by automated tests (they're not even compiled in CI).

Sure thing.  In this case, I tested the sample and it (still) works :)

**Comment by cmeeren** [comment]
> Oh, I am sorry. I thought the title said enough. Next time I will make sure to elaborate in the description.

No problem. The title described what, but not why (not in itself a problem). Without a written motivation, I just had to convince myself of why this was a good idea. It's been a while since I've looked at various parts of the code/samples. :)

File: src/Samples/SubModel/Program.fs (hunk 5 above)
Comment (bender2k14) [reply src/Samples/SubModel/Program.fs]: Yes, I prefer making it a value.  In fact, this just like by question about `bindings` #107.  I also agree with making this change in a separate PR.

Do you prefer still calling it `init`?  In my project at work, I have been calling this `initModel` (which is short for "initial model" instead of "initialize (model)", but I am not completely sold on the name.

File: src/Samples/SubModel.Views/SubModel.Views.csproj (hunk 2 above)
Comment (bender2k14) [reply src/Samples/SubModel.Views/SubModel.Views.csproj]: Oh, shoot.  I thought I had.  Nice catch.

File: src/Samples/SubModel/Program.fs (hunk 5 above)
Comment (cmeeren) [reply src/Samples/SubModel/Program.fs]: Let's stick to `init`. 👍

File: src/Samples/SingleCounter/Program.fs (hunk 1 above)
Comment (bender2k14) [reply src/Samples/SingleCounter/Program.fs]: Since the plan is to change `init` from a function to a value, I definitely prefer the resulting inline expression.  I reverted this change.

Also rebased onto newest `master`.

File: src/Samples/SubModel/Program.fs (hunk 3 above)
Comment (bender2k14) [reply src/Samples/SubModel/Program.fs]: Reverted to inline

File: src/Samples/SubModel/Program.fs (hunk 4 above)
Comment (bender2k14) [reply src/Samples/SubModel/Program.fs]: Inlined expression

**Comment by cmeeren** [comment]
Thanks! Do you mind squashing?

**Comment by bender2k14** [comment]
Gladly :)

Done.
